clusterloader2/pkg/test/simpletestexecutor.go

```diff
@@ -28,20 +33,72 @@ func createSimpleTestExecutor() TestExecutor {
 
 // ExecuteTest executes test based on provided configuration.
 func (ste *simpleTestExecutor) ExecuteTest(ctx Context, conf *api.Config) error {
+	err := ctx.GetFramework().CreateAutomanagedNamespaces(int(conf.AutomanagedNamespaces))
+	defer cleanupResources(ctx)
+	if err != nil {
+		return fmt.Errorf("automanaged namespaces creation failed: %v", err)
+	}
+
+	for i := range conf.Steps {
+		ste.ExecuteStep(ctx, &conf.Steps[i])
+	}
 	return nil
 }
 
 // ExecuteStep executes single test step based on provided step configuration.
 func (ste *simpleTestExecutor) ExecuteStep(ctx Context, step *api.Step) error {
+	if len(step.Measurements) > 0 {
+		// TODO: handle measurments
+	} else {
+		var wg wait.Group
+		defer wg.Wait()
+		for i := range step.Phases {
+			phase := &step.Phases[i]
+			wg.Start(func() {
+				ste.ExecutePhase(ctx, phase)
```

Comment:
I guess you can directly pass `&step.Phases[i]` here.

Reply:
Can't be done.
Passing `&step.Phases[i]` will make `i` to be called by inner function. `i` changes with every loop iteration, so `&step.Phases[i]` can have different values based on time which the call was made.

I can change it to `&step.Phases[j]`  and add `j :=i` instead of `phase := &step.Phases[i]`. Somewhere copy of `i` have to made so it will became "constant" for the inner function.

Reply:
You're right - I missed that sorry. Let's keep it as it is then.